### DISCUSSION TITLE
feature: Option to Associate Separate IMS Packages for GWF, GWE, GWT models in mfsim.nam

### DISCUSSION THREAD

**Discussion by MickeyRush** [modflowpy/flopy#2266] [Q&A]
The latest MODFLOW release checks for separate IMS packages in GWF-GWE simulations: 
ERROR REPORT:
  1. Problem with GWF-GWE exchange GWF-GWE_1.  The GWF model must be solved
     by a different IMS than the GWE model. Furthermore, the IMS specified
     for GWF must be listed in mfsim.nam before the IMS for GWE.

As far as I can tell, flopy does not currently have a way to associate separate IMS packages with different models, so the mfsim.nam file ends up like:
BEGIN solutiongroup  1
  ims6  ccg_gwf.ims  ccg_gwf  ccg_gwe
END solutiongroup  1

So far I have been manually editing my mfsim.nam:
BEGIN solutiongroup  1
  ims6  ccg_gwf.ims  ccg_gwf  
  ims6  ccg_gwe.ims  ccg_gwe
END solutiongroup  1

But it would be nice to have an option in the flopy.mf6.ModflowIms() or flopy.mf6.ModflowGwfgwe() to associate those separate IMS packages with different models. Apologies if it already exists and I missed it.

**Comment by wpbonelli** [comment]
@MickeyRush this can be done with `MFSimulation.register_ims_package()`, see example [here](https://flopy.readthedocs.io/en/latest/Notebooks/mf6_support_example.html).
